pkg/env/integersetting.go

```diff
@@ -45,8 +52,42 @@ func RegisterIntegerSetting(envVar string, defaultValue int) *IntegerSetting {
 	s := &IntegerSetting{
 		envVar:       envVar,
 		defaultValue: defaultValue,
+		minimumValue: math.MinInt,
+		maximumValue: math.MaxInt,
 	}
 
 	Settings[s.EnvVar()] = s
 	return s
 }
+
+// WithMinimum specifies the minimal allowed value that passes the validation.
+func (s *IntegerSetting) WithMinimum(min int) *IntegerSetting {
+	if s.defaultValue < min {
+		panic(errox.InvariantViolation.Newf("programmer error: default %d < minimum %d for %s",
+			s.defaultValue, min, s.envVar,
+		))
+	}
+	s.minimumValue = min
+	if s.minimumValue > s.maximumValue {
+		panic(errox.InvariantViolation.Newf("programmer error: incorrect validation config for %s: "+
+			"minimum value %d must be smaller or equal to maximum value %d",
+			s.EnvVar(), s.minimumValue, s.maximumValue))
+	}
+	return s
```

Comment:
What about extracting a validation function for reuse in `WithMaximum`?
```suggestion
	s.minimumValue = min
	s.mustValidate()
	return s
}

func (s *IntegerSetting) mustValidate() {
	if s.defaultValue < s.minimumValue || s.defaultValue > s.maximumValue {
		panic(errox.InvariantViolation.Newf("programmer error: default %d out of range %d .. %d for %s",
			s.defaultValue, s.minimumValue, s.maximumValue, s.envVar,
		))
	}
	if s.minimumValue > s.maximumValue {
		panic(errox.InvariantViolation.Newf("programmer error: incorrect validation config for %s: "+
			"minimum value %d must be less or equal to maximum value %d",
			s.EnvVar(), s.minimumValue, s.maximumValue))
	}
}
```

Reply:
I can, but what is the benefit? style only?

Reply:
Actually it reduces one code repetition. Here it is: [3a86da4](https://github.com/stackrox/stackrox/pull/15799/commits/3a86da411d0db59e56efda1f9b729ce7ea546038)